src/policy_metadata.rs

```diff
@@ -164,13 +165,15 @@ impl Metadata {
     }
 }
 
-fn validate_protocol_version(protocol: &ProtocolVersion) -> Result<(), ValidationError> {
-    match protocol {
-        ProtocolVersion::Unknown => Err(ValidationError::new(
+fn validate_metadata(metadata: &Metadata) -> Result<(), ValidationError> {
+    if metadata.execution_mode == PolicyExecutionMode::KubewardenWapc
+        && metadata.protocol_version == Some(ProtocolVersion::Unknown)
```

Comment:
Should we consider the case when `protocolVersion` is `None` to error out as well?

Reply:
This is already done, I've added a `#[validate(required)]` macro for it (like 125).

This crate we're using works in this way:
  1. Ensure all the field-specific validations are passing
  2. Ensure the "global" validation against the full object passes (that's the "schema function")

Reply:
That's super nice, thanks for the information!